packages/typespec-go/src/tcgcadapter/clients.ts

```diff
@@ -819,12 +819,15 @@ export class clientAdapter {
         if (go.isModelType(goType) || go.isInterfaceType(goType)) {
           let concreteType: go.ModelType | go.PolymorphicType;
           if (go.isInterfaceType(goType)) {
-            concreteType = goType.possibleTypes.find(t => t.discriminatorValue?.literal === exampleType.type.discriminatorValue)!;
+            concreteType = goType.possibleTypes.find(t => t.discriminatorValue?.literal === exampleType.type.discriminatorValue || t.discriminatorValue?.literal.value === exampleType.type.discriminatorValue)!;
           } else {
             concreteType = goType;
           }
           const ret = new go.StructExample(concreteType);
           for (const [k, v] of Object.entries(exampleType.value)) {
+            if (concreteType === undefined) {
```

Comment:
Can you please change the definition of `concreteType` to include `undefined` and move this check to before the call to `new go.StructExample(...)`.